src/PushNotifications.Delivery.Pushy/PushyTopicSubscriptionManager.cs

```diff
@@ -14,17 +16,22 @@ public PushyTopicSubscriptionManager(PushyClient pushyClient)
 
     public SubscriptionType Platform => SubscriptionType.Pushy;
 
-    public async Task<bool> SubscribeToTopicAsync(SubscriptionToken token, Topic topic)
+    public async Task<SubscribeUnsubscribeHandler> SubscribeToTopicAsync(SubscriptionToken token, Topic topic)
     {
-        bool subscribeResult = await pushyClient.SubscribeToTopicAsync(token, topic).ConfigureAwait(false);
-
-        return subscribeResult;
+        var result = await pushyClient.SubscribeToTopicAsync(token, topic).ConfigureAwait(false);
+        if (result == true)
+        {
+            return SubscribeUnsubscribeHandler.Successful();
+        }
+        return SubscribeUnsubscribeHandler.Unsuccessful(null);
```

Comment:
This is not a good approach. SubscribeUnsubscribeHandler.Unsuccessful(here accepts collection) if you pass null and somewhere in the code where you are using this result you try to enumerate over the collection it will throw an exception. It's better to pass an empty collection.